### PR TITLE
Add Chuvash layout

### LINKED ISSUE #32: Add Chuvash language layout
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Keyboard/issues) or [closed](https://github.com/FossifyOrg/Keyboard/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Keyboard/discussions) or [closed](https://github.com/FossifyOrg/Keyboard/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Keyboard/blob/main/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

Please add Chuvash Cyrillic layout. It should likely be based on the Russian one.

### Why do you want this feature?

It's my native language and I'd like to write in it.

### Additional information

Alphabet: <https://en.wikipedia.org/wiki/Chuvash_language#Official>.

### PR DESCRIPTION
**Pull request by tenextractor**
#### What is it?
- [ ] Bugfix
- [X] Feature
- [ ] Codebase improvement

#### Description of the changes in your PR
Added Chuvash layout. String "Chuvash" is hardcoded for now, I think that needs to be fixed by changing https://github.com/FossifyOrg/Commons

#### Before/After Screenshots/Screen Record
image

#### Fixes the following issue(s)

- Fixes #32 

#### Acknowledgement
- [X] I read the [contribution guidelines](https://github.com/FossifyOrg/Keyboard/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Aga-C** [comment]
> String "Chuvash" is hardcoded for now

You can quickly add it by yourself in Commons in https://github.com/FossifyOrg/Commons/blob/master/commons/src/main/res/values/strings.xml. When creating a pull request in Commons, please mention that it's for this one.

**Comment by tenextractor** [comment]
> You can quickly add it by yourself in Commons in https://github.com/FossifyOrg/Commons/blob/master/commons/src/main/res/values/strings.xml. When creating a pull request in Commons, please mention that it's for this one.
Thanks, that has been done.

**Comment by tenextractor** [comment]
When will this be merged?

**Comment by Aga-C** [comment]
Before Keyboard is released, all PRs will be merged. But I don't know when it will happen. Please be patient.

**Comment by naveensingh** [comment]
@tenextractor Do you speak these languages? I mean Chuvash, Hebrew and Esperanto.

File: app/src/main/res/xml/keys_letters_chuvash.xml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            app:keyLabel="\?"
+            app:popupCharacters="9"
+            app:popupKeyboard="@xml/keyboard_popup_template"
+            app:topSmallNumber="9" />
+        <Key
+            app:keyEdgeFlags="right"
+            app:keyLabel="&quot;"
+            app:popupCharacters="«»„“0"
+            app:popupKeyboard="@xml/keyboard_popup_template"
+            app:topSmallNumber="0" />
```
Comment (naveensingh) [inline app/src/main/res/xml/keys_letters_chuvash.xml:89]: These characters are already there in the `?123` number layout. We don't need to duplicate them here.

Also, please convert some of the less common characters into popup characters, it's always a good idea to fit all characters in 4 rows or under.

File: app/src/main/res/xml/keys_letters_chuvash.xml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        <Key
+            app:keyEdgeFlags="right"
+            app:keyLabel="э" />
+    </Row>
+    <Row>
+        <Key
+            app:code="-1"
+            app:keyEdgeFlags="left"
+            app:keyIcon="@drawable/ic_caps_outline_vector"
+            app:keyWidth="9.275%p" />
```
Comment (naveensingh) [inline app/src/main/res/xml/keys_letters_chuvash.xml:130]: Shift and Backspace should use 15% of the available space as done in other layouts.

**Review by naveensingh — CHANGES_REQUESTED** [review]

**Comment by tenextractor** [comment]
> Do you speak these languages? I mean Chuvash, Hebrew and Esperanto.

No, but I know their alphabets. The Hebrew and Esperanto layouts have been copied from Gboard, with minor changes, and the Chuvash layout is based on user feedback.

> Also, please convert some of the less common characters into popup characters, it's always a good idea to fit all characters in 4 rows or under.

I had asked the user for feedback on whether to add the example layout on the left (Й У Ӳ К Е Ӗ...), but they requested a bigger layout with all the keys visible. All of those letters are important, and we can't fit them in 4 rows. Is there any reason we need to use only 4 rows? Many languages in Gboard use 5 or more rows, like Thai for example.

**Comment by tenextractor** [comment]
> Shift and Backspace should use 15% of the available space as done in other layouts.

But this makes the keys in the bottom row narrower and harder to press accurately. If there are fewer keys in the 3rd row, bigger Shift and Backspace would be good, but both the 2nd and 3rd row have 11 keys, including Shift and Backspace.

**Comment by tenextractor** [comment]
> These characters are already there in the ?123 number layout. We don't need to duplicate them here.

The issue is that it's not feasible to add all the Chuvash letters in 4 rows, so a 5th row was added. But the 5th row only has 6 alphabet letters, so I added 4 symbols to balance the layout and allow all 10 digits to be accessible with popups.

If the digits aren't an issue, the symbols could be removed, but that would make the top row have only 6 keys.

**Comment by naveensingh** [comment]
4 rows aren't mandatory or anything, just a design choice. I think I'll just merge this now and update the layout as needed upon user feedback.

**Pull request merged**